x-pack/plugins/ml/public/application/model_management/deployment_params_mapper.ts

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { MlStartTrainedModelDeploymentRequest } from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import type { TrainedModelDeploymentStatsResponse } from '../../../common/types/trained_models';
+import type { CloudInfo } from '../services/ml_server_info';
+import type { MlServerLimits } from '../../../common/types/ml_server_info';
+import type { AdaptiveAllocations } from '../../../server/lib/ml_client/types';
+import type { DeploymentParamsUI } from './deployment_setup';
+
+export type MlStartTrainedModelDeploymentRequestNew = MlStartTrainedModelDeploymentRequest &
+  AdaptiveAllocations;
+
+const THREADS_MAX_EXPONENT = 5;
+
+type VCPUBreakpoints = Record<DeploymentParamsUI['vCPUUsage'], { min: number; max: number }>;
+/**
+ * Class responsible for mapping deployment params between API and UI
+ */
+export class DeploymentParamsMapper {
+  private readonly threadingParamsValues: number[];
+
+  /**
+   * vCPUs level breakpoints for cloud cluster with enabled ML autoscaling
+   */
+  private readonly autoscalingVCPUBreakpoints: VCPUBreakpoints = {
+    low: { min: 1, max: 1 },
+    medium: { min: 2, max: 32 },
+    high: { min: 32, max: 99999 },
+  };
+
+  /**
+   * vCPUs level breakpoints based on the ML server limits
+   */
+  private readonly hardwareVCPUBreakpoints: VCPUBreakpoints;
+
+  /**
+   * Result vCPUs level breakpoint based on the cluster env.
+   */
+  private readonly vCpuBreakpoints: VCPUBreakpoints;
+
+  constructor(
+    private readonly modelId: string,
+    private readonly mlServerLimits: MlServerLimits,
+    private readonly cloudInfo: CloudInfo,
+    private readonly showNodeInfo: boolean
+  ) {
+    const maxSingleMlNodeProcessors = this.mlServerLimits.max_single_ml_node_processors;
+
+    this.threadingParamsValues = new Array(THREADS_MAX_EXPONENT)
+      .fill(null)
+      .map((v, i) => Math.pow(2, i))
+      .filter(maxSingleMlNodeProcessors ? (v) => v <= maxSingleMlNodeProcessors : (v) => true);
+
+    const mediumValue = this.mlServerLimits!.total_ml_processors! / 2;
+
+    this.hardwareVCPUBreakpoints = {
+      low: { min: 1, max: 1 },
+      medium: { min: Math.min(2, mediumValue), max: mediumValue },
+      high: {
+        min: mediumValue,
+        max: this.mlServerLimits!.total_ml_processors!,
+      },
+    };
+
+    this.vCpuBreakpoints = this.cloudInfo.isMlAutoscalingEnabled
+      ? this.autoscalingVCPUBreakpoints
+      : this.hardwareVCPUBreakpoints;
+  }
+
+  private getNumberOfThread(input: DeploymentParamsUI): number {
```

Comment:
Nit - `getNumberOfThreads` ?

Reply:
Fixed in c5f7e84cd69e880bca8c5aad5d345ff14d31c4fd